qiskit/circuit/add_control.py

```diff
@@ -149,15 +147,43 @@ def control(operation: Union[Gate, ControlledGate],
                             use_basis_gates=True)
             elif rule[0].name == 'u1':
                 qc.mcu1(rule[0].params[0], q_control, q_target[rule[1][0].index])
-            elif rule[0].name == 'cx':
-                qc.mct(q_control[:] + [q_target[rule[1][0].index]], q_target[rule[1][1].index],
+            elif rule[0].name == 'cx' or rule[0].name == 'ccx':
+                additional_control_bits = [bit.index for bit in rule[1][:-1]]
+                qc.mct(q_control[:] + q_target[additional_control_bits],
```

Comment:
Could you use `mcx` instead of `mct`? Since the gate class itself is also called `MCXGate` I think this would be more consistent. `mct` is more like a backward compatibility alias we left there.
```suggestion
                qc.mcx(q_control[:] + q_target[additional_control_bits],
```